gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/sampling/PageRankSamplingTest.java

```diff
@@ -0,0 +1,212 @@
+   * @param sampleGreaterThanThreshold Whether to sample vertices with a PageRank-score
+   *                                   greater (true) or equal/smaller (false) the threshold
+   * @param keepVerticesIfSameScore Whether to sample all vertices (true) or none of them (false)
+   *                                in case all vertices got the same PageRank-score.
+   */
+  public PageRankSamplingTest(String testName, String seed, String sampleSize,
+    String dampeningFactor, String maxIteration, String sampleGreaterThanThreshold,
+    String keepVerticesIfSameScore) {
+    super(testName, Long.parseLong(seed), Float.parseFloat(sampleSize),
+      Double.parseDouble(dampeningFactor), Integer.parseInt(maxIteration),
+      Boolean.parseBoolean(sampleGreaterThanThreshold),
+      Boolean.parseBoolean(keepVerticesIfSameScore));
+  }
+
+  @Override
+  public SamplingAlgorithm getSamplingOperator() {
+    return new PageRankSampling(dampeningFactor, maxIteration, (double) sampleSize,
+      sampleGreaterThanThreshold, keepVerticesIfSameScore);
+  }
+
+  @Override
+  public void validateSpecific(LogicalGraph input, LogicalGraph output) {
+
+    try {
+      // test normal graph
+      GraphHead normalGh = output.getGraphHead().collect().get(0);
+      double minScore = normalGh.getPropertyValue(PageRankSampling.MIN_PAGE_RANK_SCORE_PROPERTY_KEY)
+        .getDouble();
+      double maxScore = normalGh.getPropertyValue(PageRankSampling.MAX_PAGE_RANK_SCORE_PROPERTY_KEY)
+        .getDouble();
+      if (minScore != maxScore) {
+        for (Vertex v : newVertices) {
+          assertTrue("vertex does not have scaled PageRank-score property (should have):"
+            + v.toString(), v.hasProperty(PageRankSampling.SCALED_PAGE_RANK_SCORE_PROPERTY_KEY));
+
+          if (v.hasProperty(PageRankSampling.SCALED_PAGE_RANK_SCORE_PROPERTY_KEY)) {
+            double score = v.getPropertyValue(PageRankSampling.SCALED_PAGE_RANK_SCORE_PROPERTY_KEY)
+              .getDouble();
+            if (sampleGreaterThanThreshold) {
+              assertTrue("sampled vertex has PageRankScore smaller or equal than threshold",
+                score > sampleSize);
+            } else {
+              assertTrue("sampled vertex has PageRankScore greater than threshold",
+                score <= sampleSize);
+            }
+          }
+        }
+      } else {
+        if (keepVerticesIfSameScore) {
+          assertEquals("not all vertices got sampled (should be, all got same score)",
+            newVertices.size(), dbVertices.size());
+        } else {
+          assertTrue("some vertices got sampled (should NOT be, all got same score)",
+            newVertices.isEmpty());
+        }
+      }
+      dbEdges.removeAll(newEdges);
+      for (Edge edge : dbEdges) {
+        assertFalse("there are vertices from edges, which are not part of the sampled graph",
+          newVertexIDs.contains(edge.getSourceId()) &&
```

Comment:
Not sure, but shouldn't this be a logical `or`?

Reply:
Right. Or you split it into two separate assertFalse() calls.